coredb-operator/tests/integration_tests.rs

```diff
@@ -0,0 +1,119 @@
+// It also assumes that the CRD(s) and operator are already installed for this cluster.
+// In this way, it can be used as a conformance test on a target, separate from installation.
+
+#[cfg(test)]
+mod test {
+
+    use controller::CoreDB;
+    use k8s_openapi::{
+        api::core::v1::{Namespace, Pod},
+        apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition,
+    };
+    use kube::{
+        api::{Patch, PatchParams},
+        runtime::wait::{await_condition, conditions},
+        Api, Client, Config,
+    };
+    use rand::Rng;
+
+    const API_VERSION: &str = "kube.rs/v1";
+
+    #[tokio::test]
+    #[ignore]
+    async fn functional_test_basic_create() {
+        // Initialize the Kubernetes client
+        let client = kube_client().await;
+
+        // Configurations
+        let mut rng = rand::thread_rng();
+        let name = &format!("test-coredb-{}", rng.gen_range(0..100000));
+        let namespace = "default";
+        let kind = "CoreDB";
+        let replicas = 1;
+
+        // Timeout settings while waiting for an event
+        let timeout_seconds = 30;
+
+        // Apply a basic configuration of CoreDB
+        println!("Creating CoreDB resource {}", name);
+        let coredbs: Api<CoreDB> = Api::namespaced(client.clone(), namespace);
+        let coredb_json = serde_json::json!({
+            "apiVersion": API_VERSION,
+            "kind": kind,
+            "metadata": {
+                "name": name
+            },
+            "spec": {
+                "replicas": replicas
+            }
+        });
+        let params = PatchParams::apply("coredb-integration-test");
+        let patch = Patch::Apply(&coredb_json);
+        let _coredb_resource = coredbs.patch(name, &params, &patch).await;
+
+        // Wait for Pod to be created
+
+        let pod_name = format!("{}-0", name);
+        println!("Waiting for pod to be running: {}", pod_name);
+        let pods: Api<Pod> = Api::namespaced(client.clone(), namespace);
+        if let Err(_) = tokio::time::timeout(
+            std::time::Duration::from_secs(timeout_seconds),
+            await_condition(pods, &pod_name, conditions::is_pod_running()),
+        )
+        .await
+        {
+            panic!(
+                "\n\nERROR: Did not find the pod {} to be running after waiting for {} seconds\n\n",
+                pod_name, timeout_seconds
+            )
+        }
+    }
+
+    async fn kube_client() -> kube::Client {
+        // Initialize the Kubernetes client
+        let client_future = Client::try_default();
+        let client = match client_future.await {
+            Ok(wrapped_client) => wrapped_client,
+            Err(_error) => panic!("Please configure your Kubernetes Context"),
+        };
+        // Get the name of the currently selected namespace
+        let selected_namespace = Config::infer().await.unwrap().default_namespace;
```

Comment:
NB: you are technically doing `Config::infer` twice in this block. you could use the result of the config inference to create a `Client`

Reply:
I will check on this. Thank you!